From e097a4608cafac147546db658a4a37f8a23715ca Mon Sep 17 00:00:00 2001
From: Mart Raudsepp <leio@gentoo.org>
Date: Thu, 17 Jan 2019 01:16:40 +0200
Subject: [PATCH] build: make introspection optional, not based on
 cross-compiling

This is how most other projects do it - cross-compiling that hits trouble
can just disable it then.

diff --git a/meson_options.txt b/meson_options.txt
index 9c32a5f..58e9f8c 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -6,3 +6,5 @@ option('enable-libgcrypt', type: 'combo', choices : ['yes', 'no', 'auto'], value
   description : 'Enable libgcrypt support.')
 option('enable-gtk-doc', type: 'boolean', value: 'false',
   description : 'Generate the API reference (depends on GTK-Doc)')
+option('enable-introspection', type: 'boolean', value: 'true',
+  description : 'Generate gobject-introspection bindings')
diff --git a/plparse/meson.build b/plparse/meson.build
index bdfde9d..ad4e344 100644
--- a/plparse/meson.build
+++ b/plparse/meson.build
@@ -121,7 +121,7 @@ if have_quvi
 endif
 
 # Introspection
-if not meson.is_cross_build()
+if get_option('enable-introspection')
   gnome.generate_gir(plparser_lib,
                      sources: plparser_public_headers + plparser_sources + [
                        totem_pl_parser_marshalers[1],
-- 
2.17.0

